ts/components/services/ServicesSearch.tsx

```diff
@@ -0,0 +1,168 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import React, { ComponentProps } from "react";
+
+import { SectionListData } from "react-native";
+import { ServicePublic } from "../../../definitions/backend/ServicePublic";
+import { ProfileState } from "../../store/reducers/profile";
+import { serviceContainsText } from "../../utils/services";
+import ServiceSectionListComponent from "./ServiceSectionListComponent";
+
+type OwnProps = {
+  sectionsState: // tslint:disable-next-line: readonly-array
+  Array<SectionListData<pot.Pot<ServicePublic, Error>>>;
+  searchText: string;
+  profile: ProfileState;
+  onRefresh: () => void;
+  navigateToServiceDetail: (service: ServicePublic) => void;
+};
+
+type Props = Pick<
+  ComponentProps<typeof ServiceSectionListComponent>,
+  "onRefresh"
+> &
+  OwnProps;
+
+type State = {
+  potFilteredServiceSectionsStates: pot.Pot<
+    // tslint:disable-next-line: readonly-array
+    Array<SectionListData<pot.Pot<ServicePublic, Error>>>,
+    Error
+  >;
+};
+
+/**
+ * Filter only the services that match the searchText.
+ */
+const generateSectionsServicesStateMatchingSearchTextArrayAsync = (
+  potServicesState: pot.Pot<
+    // tslint:disable-next-line: readonly-array
+    Array<SectionListData<pot.Pot<ServicePublic, Error>>>,
+    Error
+  >,
+  searchText: string
+  // tslint:disable-next-line: readonly-array
+): Promise<Array<SectionListData<pot.Pot<ServicePublic, Error>>>> => {
+  return new Promise(resolve => {
```

Comment:
why you need a Promise here? the code is not async, the promise will resolve immediately

Reply:
in the PR i fixed that